### PR TITLE
embed: support config UnaryServerInterceptor to embed server

### PR DESCRIPTION
**Pull request by Ehco1996**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

fix https://github.com/etcd-io/etcd/issues/13468

### CODE REVIEW

**Comment by Ehco1996** [comment]
@hexfusion  ptal 😬

**Comment by CabinfeverB** [comment]
It seems a good solution to support adding ServerOption for stream and unary interceptor

File: server/embed/config.go [hunk 1]
```diff
@@ -254,6 +254,9 @@ type Config struct {
 	// SocketOpts are socket options passed to listener config.
 	SocketOpts transport.SocketOpts `json:"socket-options"`
 
+	// GRPCInterceptor specifies the grpc server interceptors.
+	GRPCInterceptor grpc.UnaryServerInterceptor
```
Comment (nolouch) [inline server/embed/config.go:258]: How about export ability to add customized `grpc.ServerOption`. like grpcCustomOptions []gprc.ServerOption

Comment (Ehco1996) [reply server/embed/config.go:258]: sgtm, but i find hard to dynamic change `grpc.ServerOption` to `grpc.UnaryServerInterceptor`
`impossible type switch case: opt (type grpc.ServerOption) cannot have dynamic type grpc.UnaryServerInterceptor (missing grpc.apply method)`

could you give me some advice for this ? @nolouch

**Comment by Ehco1996** [comment]
closed for now, i will try better ways later

**Pull request closed without merging**